src/OrchardCore/OrchardCore.ContentManagement.Display/ContentDisplayManager.cs

```diff
@@ -170,6 +171,26 @@ public async Task<IShape> UpdateEditorAsync(ContentItem contentItem, IUpdateMode
 
             await _contentHandlers.InvokeAsync((handler, updateContentContext) => handler.UpdatingAsync(updateContentContext), updateContentContext, _logger);
             await _handlers.InvokeAsync((handler, contentItem, context) => handler.UpdateEditorAsync(contentItem, context), contentItem, context, _logger);
+            var validateContentContext = new ValidateContentContext(updateContentContext.UpdatingItem);
+            await _contentHandlers.InvokeAsync((handler, validateContentContext) => handler.ValidatingAsync(validateContentContext), validateContentContext, _logger);
+            await _contentHandlers.InvokeAsync((handler, validateContentContext) => handler.ValidatedAsync(validateContentContext), validateContentContext, _logger);
+            if (!validateContentContext.ContentValidateResult.Succeeded)
+            {
+                foreach (var error in validateContentContext.ContentValidateResult.Errors)
+                {
+                    if (error.MemberNames.Any())
+                    {
+                        foreach (var member in error.MemberNames)
+                        {
+                            context.Updater.ModelState.AddModelError(context.HtmlFieldPrefix, member, error.ErrorMessage);
```

Comment:
However this line is not just for `AliasPart `or `AutoroutePart` this could be for every Parts and Fields - Orchard provided or Custom module provided and having just `nameof()` may show errors in ValidationSummary but would enough to represent accurate field in UI editor.

For example if the part is consist of complex properties and not just primitive types, e.g `AddressPart` as following

   ```
 public class AddressPart : ContentPart
    {
        public string Street {get;set;}
        public State State {get;set;}
        public Country Country {get;set;}
    }
    public class State 
    {
        public string Abbr {get;set;}
        public string Name {get;set;}
    }
    public class Country 
    {
        public string Abbr {get;set;}
        public string Name {get;set;}
    }
```

using member name with `nameof(State.Name)`  and `nameof(Country.Name)`would generate same key `Name` the correct name should be `State.Name` and `Country.Name`

Reply:
@ns8482e thanx for the clarification, I see the point. But IMO we should handle how to add the name of the part as a prefix among with the html Prefix used in the driver in the validation context. In the example, State and Country should be able to validate themselves and in the validating handler the developer should add State.ValidationErrors and Country.ValidationErrors prefixed by partName. If we prefer the penalty of creating an expression and traversing the tree in order to find the name for each validation for giving an API to resolve nested object names, we could change it.